### DISCUSSION TITLE
admin doesn't seem to work with intermediate join table for manytomany

### DISCUSSION THREAD

**Discussion by bryandunbar** [jazzband/django-polymorphic#808] [Q&A]
I have many to many that uses `through` and specifies a join table. The admin integration for Polymorphic inlines doesn't seem to function. Is this a known issue, or am I missing something?

**Comment by diogosimao** [comment]
Do you get error:

> 'ManyToManyField' object has no attribute '_m2m_reverse_name_cache'

...?

**Comment by JohananOppongAmoateng** [comment]
This is **not a bug**.
`django-polymorphic` does **not support automatic admin inlines for `ManyToManyField` with a custom `through` model**.

Django admin only works with **explicit inlines on the through model**, and polymorphic admin doesn’t add extra support on top of that.

**What to do:**

* Define a normal `TabularInline` / `StackedInline` for the **through model**
* Attach it manually to the `PolymorphicChildModelAdmin`

**Comment by JohananOppongAmoateng** [comment]
@bckohan I think this can be made into a discussion too

**Comment by bckohan** [comment]
@bryandunbar @diogosimao This works, but you have to inherit an inline admin class for the through model from StackedPolymorphicInline:

```python

class A(PolymorphicModel):
    ...

class B(A):
    ...

class AThrough(PolymorphicModel):
    ...

class BThrough(AThrough):  # through models can be polymorphic too!
    ...

class YourModel(models.Model):

    m2m_field = models.ManyToManyField(A, through=AThrough, related_name="my_models", blank=True)

```

In your admin:

```python

import polymorphic.admin.StackedPolymorphicInline, PolymorphicInlineSupportMixin

class AThroughMembership(StackedPolymorphicInline):

    model = AThrough
    extra = 1

    class AThroughInline(StackedPolymorphicInline.Child):
        ...

        model = AThrough

    class BThroughInline(StackedPolymorphicInline.Child):
        ...

        model = BThrough

    child_inlines = (AThroughInline, BThroughInline)


@register(YourModel)
class YourModelAdmin(PolymorphicInlineSupportMixin, PolymorphicChildModelAdmin):

    inlines = (AThroughMembership,)

```

You can see a full working example in the tests added in #807
